cloud_testing/xctest_parser/src/test/kotlin/xctest/ParseTest.kt

```diff
@@ -47,6 +51,45 @@ class ParseTest {
             "EarlGreyExampleSwiftTests/testWithInRoot"
     )
 
+    init {
+        val assetLink = remoteAssetLink()
+        if (updateRequired(assetLink)) {
+            updateFixtures(assetLink)
+        }
+    }
+
+    private fun remoteAssetLink(): Element {
+        val doc = Jsoup.connect("https://github.com/Flank/test_artifacts/releases/tag/v0.1").get()
+        val downloadLinks = doc.select("li.d-block > a")
+        return downloadLinks.find {
+            it.attr("href").endsWith(".zip") && it.attr("href").contains("releases/download")
+        } ?: throw RuntimeException("Download link not found in html!")
+    }
+
+    private fun updateRequired(remoteAssetLink: Element): Boolean {
+        val remoteAssetName = remoteAssetLink.attr("href").split("/").last()
+        val fixtures = File(fixturesPath)
+        if (!fixtures.exists()) return true
+        val localAssetPath = fixtures.listFiles().find { it.extension == "zip" } ?: return true
+        val localAssetMd5 = localAssetPath.readBytes().md5()
```

Comment:
the md5 should be the file name, this avoids us having to recalculate the md5 for each test run which may be expensive for large zips.

Reply:
I think the value of testing is in developer productivity so we'd want to optimize the feedback loop to be as performant as possible.

I don't think zip corruption is significant enough to slow down every test run. A faster way would be to check the file size. If the remote file size matches the local, then there's less chance of any corruption.